test/codec.ts

```diff
@@ -631,6 +659,48 @@ describe('codec', () => {
 
       assert.strictEqual(encoded, 10);
     });
+
+    it('should encode JSON', () => {
+      const expected = '{"result":true,"count":42}';
+      const value = JSON.parse('{"result": true, "count": 42}');
+
+      const encoded = codec.encode(value);
+
+      assert.deepStrictEqual(encoded, expected);
+    });
+
+    it('should encode complex object as JSON', () => {
+      const value = {
+        boolKey: true,
+        numberKey: 3.14,
+        stringKey: 'test',
+        objectKey: {innerKey: 'inner-value'},
+      };
+
+      const encoded = codec.encode(value);
+
+      assert.deepStrictEqual(
+        encoded,
+        '{"boolKey":true,"numberKey":3.14,"stringKey":"test","objectKey":{"innerKey":"inner-value"}}'
+      );
+    });
+
+    it('should encode deeply-nested object as JSON', () => {
```

Comment:
Should we also add a case for decoding a deeply-nested object or is that overkill?

Reply:
I think it's reasonable to have it symmetrical, so I've added a test for that as well.